pingtest.go

```diff
@@ -40,6 +41,45 @@ func (p *pingTest) Interval() time.Duration {
 	return p.interval
 }
 
+func AllowedIPAddress(imup *imup) (bool, error) {
+	allowed := true
+	blocked := false
+
+	if len(imup.cfg.AllowedIPs()) > 0 || len(imup.cfg.BlockedIPs()) > 0 {
+		if ip, err := util.PublicIP(); err != nil {
+			log.Error(err)
+			imup.Errors.write("IdentifyPublicIP", err)
+		} else {
+			// iterate over list of allowed ips and ensure the public ip is a match
+			for _, v := range imup.cfg.AllowedIPs() {
+				if ip == v {
+					allowed = true
```

Comment:
```suggestion
					return true, nil
```

this would make it more efficient/equivalent to what speed test is currently doing